mempool/src/tests/integration_tests.rs

```diff
@@ -150,6 +154,172 @@ async fn vfn_middle_man_test() {
+        .await;
+    node.verify_broadcast_and_ack(other_peer_network_id, &all_txns[0..1])
+        .await;
+}
+
+/// Tests when a node gets disconnected.  Node should pick up after the second sending
+/// TODO: also add an outbound test to ensure it'll broadcast all transactions again
+#[tokio::test]
+async fn test_interrupt_in_sync_inbound() {
+    let all_txns = test_transactions(0, 2);
+    let all_txns = all_txns.as_slice();
+    for (mut node, (other_peer_network_id, other_metadata)) in inbound_node_combinations() {
+        // After we connect, partial should make progress
+        node.connect_self(other_peer_network_id.network_id(), other_metadata.clone());
+        node.send_message(
+            ProtocolId::MempoolDirectSend,
+            other_peer_network_id,
+            &all_txns[0..1],
+        )
+        .await;
+        node.assert_only_txns_in_mempool(&all_txns[0..1]);
+
+        // Drop the connection, and a reconnect should merge txns
+        node.disconnect_self(other_peer_network_id.network_id(), other_metadata.clone());
+
+        node.send_message(
+            ProtocolId::MempoolDirectSend,
+            other_peer_network_id,
+            &all_txns[0..2],
+        )
+        .await;
+        node.assert_only_txns_in_mempool(&all_txns[0..2]);
+    }
+}
+
+/// Tests that transactions will only be sent when they're ready (previous seq no have been sent)
+#[tokio::test]
+async fn test_ready_txns() {
+    let all_txns = test_transactions(0, 2);
+    let all_txns = all_txns.as_slice();
+    let mut node = MempoolTestFrameworkBuilder::single_validator();
+    let (other_peer_network_id, other_metadata) =
+        validator_mock_connection(ConnectionOrigin::Inbound, &ALL_PROTOCOLS);
+
+    // Add transactions
+    node.assert_txns_not_in_mempool(&all_txns[1..2]);
+    node.add_txns_via_client(&all_txns[1..2]).await;
+
+    // After we connect, partial should make progress
+    node.connect_self(other_peer_network_id.network_id(), other_metadata.clone());
+    node.assert_txns_not_in_mempool(&all_txns[0..2]);
+    node.wait_for_no_msg(
+        other_peer_network_id.network_id(),
+        Duration::from_millis(100),
+    )
+    .await;
+
+    // Adding earlier txns should fill in the gaps, and make it send all
+    node.add_txns_via_client(&all_txns[0..1]).await;
+    node.assert_txns_in_mempool(&all_txns[0..2]);
+    node.verify_broadcast_and_ack(other_peer_network_id, &all_txns[0..2])
+        .await;
+    node.assert_only_txns_in_mempool(&all_txns[0..2]);
+}
+
+/// Test that in the validator network, messages won't be sent back to the original sender
+#[tokio::test]
+async fn test_broadcast_self_txns() {
+    let all_txns = test_transactions(0, 2);
+    let all_txns = all_txns.as_slice();
+    let mut node = MempoolTestFrameworkBuilder::single_validator();
+    let (other_peer_network_id, other_metadata) =
+        validator_mock_connection(ConnectionOrigin::Inbound, &ALL_PROTOCOLS);
+
+    // Other node sends txns
```

Comment:
This comment confuses me a little. It seems to contradict the comment below where we use `node.send_message` again.

I wonder if there's a way to make it clearer? e.g., have a `node.send_message` and a `node.receive_message` as ways of indicating that one is sending from the node and one is coming to the node?